tests/numerics/test_time_step_control.py

```diff
@@ -348,6 +349,36 @@ def test_number_of_recomp_attempts_greater_than_zero(
             )
         assert msg in str(excinfo.value)
 
+    @pytest.mark.parametrize(
+        "num_time_steps, final_time",
+        [
+            (1, 1.0e-14),
+            (3, 1.0e-14),
+            (1, 1.0),
+            (3, 1.0),
```

Comment:
As I understand it, the problem we are trying to deal with is related to rounding errors. Perhaps contradicting my previous remark regarding these parameters, I think the number of time steps should be set to maximize the chance for such errors. In that regard, 1 is a bad number, 3 is good, and so would, say, 7 be. In the same spirit, would it make sense to change the final time, say, to `pi`?

Reply:
See comment above.

Reply:
@jwboth: Agree. Ping me when you have implemented, then I will approve.